pkg/cli/image/extract/extract.go

```diff
@@ -311,6 +316,9 @@ func (o *ExtractOptions) Validate() error {
 	if len(o.Mappings) == 0 {
 		return fmt.Errorf("you must specify one or more paths or files")
 	}
+	if o.FilterOptions.IsWildcardFilter() {
+		return fmt.Errorf("you must specify a single os/arch with --filter-by-os=os/arch, wildcards cannot be used with extract")
```

Comment:
That's not really true (if wildcards filter to a single image it's fine), and we don't do this in other contexts like image info.  I don't think this error should be here.

Reply:
with `quay.io/prometheus/prometheus:v2.18.1` as an example:
`oc image info` gives a nice error msg with the list of os/arch choices available for the image.  `oc image extract` gives the confusing: `error: unable to parse image quay.io/prometheus/prometheus:v2.18.1: unknown image manifest of type *manifestlist.DeserializedManifestList`

Reply:
er, that ugly error msg is only produced if a user passes the wildcard filter-by-os, see here:
```console
$ oc image extract quay.io/prometheus/prometheus:v2.18.1 --filter-by-os=foo
error: unable to read image quay.io/prometheus/prometheus:v2.18.1: filtered all images from manifest list
$ oc image extract quay.io/prometheus/prometheus:v2.18.1 --filter-by-os=./*
error: unable to parse image quay.io/prometheus/prometheus:v2.18.1: unknown image manifest of type *manifestlist.DeserializedManifestList from manifest sha256:5880ec936055fad18ccee798d2a63f64ed85bd28e8e0af17c6923a090b686c3d
```
and also, if no filter-by-os passed then ProcessManifestList does its thing.  There's a slight room for improvement here, but yea, my original plan of erroring w/ wildcard filter is wrong, i'll update that.

Reply:
updated. PlatformSpec is set in release/new and passed to extract, to be used to inform user w/ better error msg when passing an unavailable os/arch like --filter-by-os=foo/foo  or if --filter-by-os=.* but manifests in manifestLIst >1, then print the ImageRetriever error msg that tells user to choose an os/arch.